homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -59,9 +63,15 @@
 
 CLIMATE_SCHEMA = vol.Schema(
     {
-        vol.Optional(CONF_PRECISION): vol.In(
+        vol.Optional(CONF_READ_PRECISION): vol.In(
             [PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE]
         ),
+        vol.Optional(CONF_SET_PRECISION): vol.In(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
The integration fully supports config flows and entries, this schema is a leftover from before that. At the moment it's in place to support upgrades from *really* old versions, but I agree we should not add to this.

Related question: how long are we supposed to keep supporting these kinds of cases? i.e. when can this be removed entirely?

Reply:
Our normal deprecation period is 2 release cycles, ie two months.

Reply:
So should I revert the schema to how it was? Or remove it entirely? It wouldn't be compatible with my changes anyway

Reply:
Please revert it to the existing schema.

Reply:
Done :)
Thanks for the feedback and sorry for the mess I'm making, it's my first time committing something for Home Assistant and do have to learn a thing or two apparently